### PR TITLE
DSD-1228: TextInput accessibility updates

### PR DESCRIPTION
**Pull request by oliviawongnyc**
Fixes JIRA ticket [DSD-1228.](https://jira.nypl.org/browse/DSD-1228)

## This PR does the following:

- Adds `autocomplete` attribute to `TextInput`s with type `"tel"`, `"email"`, and `"url"`.
- Fixes issue where `aria-describedby` attribute was not being added to some `TextInput`s.

## How has this been tested?

- Locally in Storybook, and with a new unit test.

## Accessibility concerns or updates

- None.

### Checklist:

- [x] I have updated the Storybook documentation accordingly.
- [x] I have added relevant accessibility documentation for this pull request.
- [x] All new and existing tests passed.

### Front End Review:

- [ ] Review the Vercel preview deployment once it is ready.

### CODE REVIEW

File: src/components/TextInput/TextInput.stories.mdx [hunk 1]
```diff
@@ -53,7 +53,7 @@ import DSProvider from "../../theme/provider";
 | Component Version | DS Version |
 | ----------------- | ---------- |
 | Added             | `0.22.0`   |
-| Latest            | `1.3.1`    |
+| Latest            | `1.3.2`    |
```
Comment (EdwinGuzman) [inline src/components/TextInput/TextInput.stories.mdx]: Make this 1.4.0 so we can bundle it in the next release.

**Review by EdwinGuzman — COMMENTED** [review]
I don't see the `autocomplete` attribute in the input element, but I should, right?

File: src/components/TextInput/TextInput.stories.mdx [hunk 2]
```diff
@@ -138,6 +141,10 @@ When the `type` prop is set to `"textarea"`, the `<textarea>` element
 is rendered instead of the `<input>` element. This element follows all the same
 accessibility rules described above.
 
+If the `type` prop is set to `"email"`, `"tel"`, or `"url"`, the appropriate
```
Comment (bigfishdesign13) [inline src/components/TextInput/TextInput.stories.mdx:144]: To support this info, add another link to the `Resources` list:

MDN HTML attribute: autocomplete
https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/autocomplete

**Review by bigfishdesign13 — CHANGES_REQUESTED** [review]
Everything seems to be working well for me.  This PR just needs one minor addition (my comment) and one update (Edwin's comment).

@EdwinGuzman I am seeing the `autocomplete` attribute as expected.  Are you looking at a variant other than `email`, `tel` or `url`?  Only those three will show the attribute by default.

**Comment by EdwinGuzman** [comment]
I do see the attribute now so all is good on my end.

**Review by EdwinGuzman — APPROVED** [review]

**Review by bigfishdesign13 — APPROVED** [review]
Yes.

**Pull request merged**